plugins/inputs/nats_consumer/nats_consumer.go

```diff
@@ -82,16 +93,34 @@ func (n *natsConsumer) SetParser(parser parsers.Parser) {
 }
 
 func (n *natsConsumer) natsErrHandler(c *nats.Conn, s *nats.Subscription, e error) {
-	select {
-	case n.errs <- natsError{conn: c, sub: s, err: e}:
-	default:
-		return
+	n.Log.Errorf("%s url:%s id:%s sub:%s queue:%s", e.Error(), c.ConnectedUrl(), c.ConnectedServerId(), s.Subject, s.Queue)
+}
+
+func (n *natsConsumer) Init() error {
+	n.messages = map[telegraf.TrackingID]bool{}
+	n.parsingConfig = parsing.NewConfig(transformConfig(n.SubjectParsing), "subject", ".", "*", n.Log)
+	return n.parsingConfig.Init()
+}
+
+func transformConfig(subjectParsing []SubjectParsingConfig) []parsing.ConfigEntry {
+	ret := make([]parsing.ConfigEntry, len(subjectParsing))
+	for i, c := range subjectParsing {
+		ret[i] = parsing.ConfigEntry{
+			Base:        c.Base,
+			Measurement: c.Measurement,
+			Tags:        c.Tags,
+			Fields:      c.Fields,
+			FieldTypes:  c.FieldTypes,
+		}
 	}
+	return ret
 }
 
 // Start the nats consumer. Caller must call *natsConsumer.Stop() to clean up.
 func (n *natsConsumer) Start(acc telegraf.Accumulator) error {
 	n.acc = acc.WithTracking(n.MaxUndeliveredMessages)
+	n.ctx, n.cancel = context.WithCancel(context.Background())
```

Comment:
You should not store the context in structs. See https://go.dev/blog/context-and-structs.